test/e2e/node_feature_discovery_test.go

```diff
@@ -899,6 +926,39 @@ denyLabelNs: []
 
 })
 
+// simplyAndStupidRetryLoop is a simple and stupid re-try loop
+func simplyAndStupidRetryLoop(poll func() error) error {
```

Comment:
nit: I'd suggest to rename this to smth more appropriate 😅

```suggestion
// simplePoll is a simple and stupid re-try loop
func simplePoll(poll func() error) error {
```